sormas-backend/src/test/java/de/symeda/sormas/backend/immunization/ImmunizationFacadeEjbPseudonymizationTest.java

```diff
@@ -0,0 +1,217 @@
+package de.symeda.sormas.backend.immunization;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.emptyString;
+import static org.hamcrest.Matchers.is;
+
+import java.util.HashMap;
+
+import javax.persistence.Query;
+
+import org.junit.jupiter.api.Test;
+
+import de.symeda.sormas.api.Disease;
+import de.symeda.sormas.api.caze.CaseDataDto;
+import de.symeda.sormas.api.feature.FeatureConfigurationIndexDto;
+import de.symeda.sormas.api.feature.FeatureType;
+import de.symeda.sormas.api.feature.FeatureTypeProperty;
+import de.symeda.sormas.api.immunization.ImmunizationDto;
+import de.symeda.sormas.api.immunization.ImmunizationManagementStatus;
+import de.symeda.sormas.api.immunization.ImmunizationStatus;
+import de.symeda.sormas.api.immunization.MeansOfImmunization;
+import de.symeda.sormas.api.person.PersonDto;
+import de.symeda.sormas.api.user.DefaultUserRole;
+import de.symeda.sormas.api.user.UserDto;
+import de.symeda.sormas.api.utils.DataHelper;
+import de.symeda.sormas.backend.AbstractBeanTest;
+import de.symeda.sormas.backend.TestDataCreator;
+import de.symeda.sormas.backend.feature.FeatureConfiguration;
+
+public class ImmunizationFacadeEjbPseudonymizationTest extends AbstractBeanTest {
+
+	private TestDataCreator.RDCF rdcf1;
+	private TestDataCreator.RDCF rdcf2;
+
+	private UserDto districtUser1;
+	private UserDto nationalUser;
+
+	@Override
+	public void init() {
+		super.init();
+		rdcf1 = creator.createRDCF("Region 1", "District 1", "Community 1", "Facility 1", "Point of entry 1");
+		rdcf2 = creator.createRDCF("Region 2", "District 2", "Community 2", "Facility 2", "Point of entry 2");
+		nationalUser = creator.createUser(
+			rdcf1.region.getUuid(),
+			rdcf1.district.getUuid(),
+			rdcf1.community.getUuid(),
+			rdcf1.facility.getUuid(),
+			"Nat",
+			"User",
+			creator.getUserRoleReference(DefaultUserRole.NATIONAL_USER));
+
+		districtUser1 = creator.createUser(
+			rdcf1.region.getUuid(),
+			rdcf1.district.getUuid(),
+			rdcf1.facility.getUuid(),
+			"Surv",
+			"Off1",
+			creator.getUserRoleReference(DefaultUserRole.SURVEILLANCE_OFFICER));
+	}
+
+	@Test
+	public void testPseudonymized() {
+		loginWith(nationalAdmin);
+
+		// immunization within limited user's jurisdiction
+		PersonDto person1 = creator.createPerson("John", "Doe");
+		final ImmunizationDto immunization1 = creator.createImmunization(
+			Disease.CORONAVIRUS,
+			person1.toReference(),
+			nationalUser.toReference(),
+			ImmunizationStatus.ACQUIRED,
+			MeansOfImmunization.VACCINATION,
+			ImmunizationManagementStatus.COMPLETED,
+			rdcf1);
+		immunization1.setAdditionalDetails("confidential details");
```

Comment:
please use the creator method with a consumer so you don't need to update the object right after creation

Reply:
changed the creator method to the one that has a consumer